src/Network/Transport/TCP/Internal.hs

```diff
@@ -113,16 +121,18 @@ data ConnectionRequestResponse =
 
 decodeConnectionRequestResponse :: Word32 -> Maybe ConnectionRequestResponse
 decodeConnectionRequestResponse w32 = case w32 of
-  0 -> Just ConnectionRequestAccepted
-  1 -> Just ConnectionRequestInvalid
-  2 -> Just ConnectionRequestCrossed
-  _ -> Nothing
+  0xFFFFFFFF -> Just ConnectionRequestUnsupportedVersion
```

Comment:
This message is version-independent, the others are not. I wanted the code to indicate that. I suppose `ConnectionRequestUnsupportedVersion` should not even be in the same datatype as the others.